src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -401,6 +421,37 @@ private void ensureSelectedMedHistIsValid(ListChangeListener.Change<? extends Me
         }
     }
 
+    /**
+     * Ensures {@code selectedAppointment} is a valid appointment in {@code filteredAppointments}.
```

Comment:
neat and clean logic that ensures a selected appointment. However it would be better to provide some test cases.